packages/amplify_core/lib/src/types/exception/api/operation_error.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/src/types/exception/api/api_exception.dart';
+
+/// {@template amplify_core.api.api_operation_error}
+/// An in-process operation encountered a processing error
+/// {@endtemplate}
+class ApiOperationError extends ApiException {
```

Comment:
nit/question: Should this be named `ApiOperationException`? as it seems most our exceptions end w `*Exception` though I know this is following swift.

Reply:
"Error" has a very specific meaning in Dart and is for classes extending `Error` which shouldn't be caught. Recommend `Exception` as well.